src/win/pipe.c

```diff
@@ -2031,6 +2031,32 @@ int uv_pipe_pending_count(uv_pipe_t* handle) {
 }
 
 
+int uv_pipe_getsockname(const uv_pipe_t* handle, char* buffer, size_t* size) {
+  if (handle->flags & UV_HANDLE_BOUND)
+    return uv__pipe_getname(handle, buffer, size);
+
+  if (handle->flags & UV_HANDLE_CONNECTION ||
+      handle->handle != INVALID_HANDLE_VALUE) {
+    *size = 0;
+    return 0;
+  }
+
+  return UV_EBADF;
+}
+
+
+int uv_pipe_getpeername(const uv_pipe_t* handle, char* buffer, size_t* size) {
+  /* emulate unix behaviour */
+  if (handle->flags & UV_HANDLE_BOUND)
```

Comment:
Really? Or do you mean

~~~~
if (!(handle->flags & UV_HANDLE_BOUND))
~~~~

?